python/plugins/processing/algs/gdal/ogr2ogrtopostgis.py

```diff
@@ -160,8 +162,8 @@ def getConsoleCommands(self):
         ssrs = unicode(self.getParameterValue(self.S_SRS))
         tsrs = unicode(self.getParameterValue(self.T_SRS))
         asrs = unicode(self.getParameterValue(self.A_SRS))
-        host = unicode(self.getParameterValue(self.HOST))
-        port = unicode(self.getParameterValue(self.PORT))
+        host = unicode(self.getParameterValue(self.HOST)) or os.environ.get('PGHOST')
+        port = unicode(self.getParameterValue(self.PORT)) or os.environ.get('PGPORT')
         user = unicode(self.getParameterValue(self.USER))
         dbname = unicode(self.getParameterValue(self.DBNAME))
```

Comment:
or os.environ.get('PGDATABASE') or user

.. or, if allowed, completely omit from he OGR connection string, then libpq would do the right thing